runtimes/runtimes/lsp/router/loggingServer.ts

```diff
@@ -0,0 +1,45 @@
+import { InitializeParams, InitializeResult, Logging } from '../../../server-interface'
+import { LspServer } from './lspServer'
+import { Connection } from 'vscode-languageserver/node'
+import { Encoding } from '../../encoding'
+import { DEFAULT_LOG_LEVEL, isValidLogLevel, DefaultLogger, LogLevel } from '../../util/loggingUtil'
+
+export class LoggingServer {
+    private logger: DefaultLogger
+    private lspServer: LspServer
+
+    constructor(
+        private lspConnection: Connection,
+        private encoding: Encoding
+    ) {
+        this.logger = new DefaultLogger(DEFAULT_LOG_LEVEL as LogLevel, this.lspConnection)
+        this.lspServer = new LspServer(this.lspConnection, this.encoding, this.logger)
+        this.lspServer.setInitializeHandler(async (params: InitializeParams): Promise<InitializeResult> => {
+            this.updateLoggingLevel(params.initializationOptions?.logLevel ?? ('log' as LogLevel))
+            return {
+                capabilities: {},
+            }
+        })
+        this.lspServer.setDidChangeConfigurationHandler(async params => {
+            const logLevelConfig = await lspConnection.workspace.getConfiguration({
+                section: 'aws.logLevel',
+            })
+            if (isValidLogLevel(logLevelConfig)) {
+                this.updateLoggingLevel(logLevelConfig as LogLevel)
+            }
+        })
+    }
+
+    private updateLoggingLevel(logLevel: LogLevel) {
+        this.logger.level = logLevel
+        this.logger.info(`Logging level changed to ${logLevel}`)
```

Comment:
We should log this only if value really changed. I think it's possible that other config value changed that triggers `setDidChangeConfigurationHandler` and then we log this, which could be misleading.